packages/fluentui/react-builder/src/components/ReaderNarration.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { Alert, Ref, Dropdown, DropdownProps } from '@fluentui/react-northstar';
+import { IAriaElement } from './../narration/NarrationComputer';
+import { DescendantsNarrationsComputer } from './../narration/DescendantsNarrationsComputer';
+
+const computer: DescendantsNarrationsComputer = new DescendantsNarrationsComputer();
+let narrationTexts: Record<string, string> = {};
+
+export type ReaderNarrationProps = {
+  selector: string;
+};
+
+export const ReaderNarration: React.FunctionComponent<ReaderNarrationProps> = ({ selector }) => {
+  const ref = React.useRef<HTMLElement>();
+  const [narrationText, setNarrationText] = React.useState('');
+  const [narrationPath, setNarrationPath] = React.useState('');
+  const [narrationPaths, setNarrationPaths] = React.useState([]);
+
+  React.useEffect(() => {
+    if (ref.current) {
```

Comment:
Might be personal preference, but when there is an if block that spans all your logic I think it's cleaner to invert and return early

Reply:
```suggestion
    if (!ref.current) {
      return;
```

Reply:
Agree. Thanks.